### DISCUSSION TITLE
terraform flux v2 with azure devops

### DISCUSSION THREAD

**Discussion by MohamedBenighil** [fluxcd/flux2#3090] [Q&A]
Hello,

a couple of days ago, i confgure flux v2 to run with azure devops by following this link https://fluxcd.io/flux/use-cases/azure/#flux-installation-for-azure-devops

However, today i would like to do exactly the same thing using Terraform this time. I know there is a flux provider : https://registry.terraform.io/providers/fluxcd/flux/latest/docs. But i don't know how to adapt it to azure devops

 Do you have any documentation about that ?

Thank you for your help

**Comment by kingdonb** [comment]
Hello! There is no specific documentation for Azure DevOps in Flux's terraform provider docs, however you should be able to use the Azure DevOps provider together with the Flux provider, if you have understood how it works

> `resource "gitlab_repository_file" "sync" {`

^ This resource makes a Git repository file in GitLab (eg. commits the file to the git repository), and you can do a similar thing with the Azure DevOps Provider:

https://github.com/microsoft/terraform-provider-azuredevops/blob/main/website/docs/r/git_repository_file.html.markdown

These are comparable. Both providers take the input of a file, which is what Flux provider outputs. Hope this helps!
